src/test/java/uk/gov/pay/api/utils/mocks/ConnectorMockClient.java

```diff
@@ -289,6 +297,20 @@ public void respondOk_whenCreateCharge(String chargeTokenId, String gatewayAccou
                         .withHeader(LOCATION, chargeLocation(gatewayAccountId, responseFromConnector.getChargeId()))
                         .withBody(buildChargeResponse(build)));
     }
+    
+    public void respondOk_whenCreateAgreement(String gatewayAccountId, CreateAgreementRequestParams requestParams) {
+        var responseFromConnector = aCreateAgreementResponseFromConnector()
+                .withReference(requestParams.getReference())
+                .withAgreementId("12345678901234567890123456")
```

Comment:
This should probably be in a public constant with that constant reused in the tests that rely on it.